### ISSUE TITLE
Failure to pull private images from Docker Hub

### ISSUE THREAD

**Issue by bkeroackdsc** [GoogleCloudPlatform/kubernetes#8012]
Pods containing public container images (eg: "redis:latest") work normally however when I try to use private images on Docker Hub I get messages like this in the journal:

```
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="GET /images/dollarshaveclub/docker-registry-v1:latest/json"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="+job image_inspect(dollarshaveclub/docker-registry-v1:latest)"
May 09 00:05:19 node.domain.com dockerd[14555]: No such image: dollarshaveclub/docker-registry-v1:latest
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="-job image_inspect(dollarshaveclub/docker-registry-v1:latest) = ERR (1)"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=error msg="Handler for GET /images/{name:.*}/json returned error: No such image: dollarshaveclub/docker-registry-v1:latest"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=error msg="HTTP Error: statusCode=404 No such image: dollarshaveclub/docker-registry-v1:latest"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="POST /images/create?fromImage=dollarshaveclub%2Fdocker-registry-v1&tag=latest"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="+job pull(dollarshaveclub/docker-registry-v1, latest)"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="+job resolve_repository(dollarshaveclub/docker-registry-v1)"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="-job resolve_repository(dollarshaveclub/docker-registry-v1) = OK (0)"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="GET /containers/json"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="+job containers()"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="-job containers() = OK (0)"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="GET /containers/json"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="+job containers()"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="-job containers() = OK (0)"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="GET /containers/json"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="+job containers()"
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="-job containers() = OK (0)"
May 09 00:05:19 node.domain.com dockerd[14555]: Error: image dollarshaveclub/docker-registry-v1:latest not found
May 09 00:05:19 node.domain.com dockerd[14555]: time="2015-05-09T00:05:19Z" level=info msg="-job pull(dollarshaveclub/docker-registry-v1, latest) = ERR (1)"
May 09 00:05:19 node.domain.com kubelet[16481]: I0509 00:05:19.743485   16481 event.go:200] Event(api.ObjectReference{Kind:"Pod", Namespace:"default", Name:"docker.registry-maq79", UID:"43901f71-f5de-11e4-99da-02cf05b1026b", APIVersion:"v1beta3", ResourceVersion:"5705", FieldPath:"spec.containers{docker-registry-v1}"}): reason: 'failed' Failed to pull image "dollarshaveclub/docker-registry-v1:latest": Error: image dollarshaveclub/docker-registry-v1:latest not found
```

I've put valid credentials in ``/home/core/.dockercfg`` per the documentation. If I do a manual ``docker pull`` on the command line on the nodes it works fine. In fact, after manually pulling the images the pod will then start up and run normally. It seems like k8s cannot pull the images itself for some reason.

I'm running v0.15.0 on CoreOS alpha (668.2.0) on AWS.

Thanks for any help

**Comment by cvallance** [comment]
So just to be sure, `sudo ls -ld /proc/$(pidof kubelet)/cwd` returned `/home/core`?

**Comment by bkeroackdsc** [comment]
Hmm. Appears not:

```bash
core@ip-x-x-x-x ~ $ sudo ls -ld /proc/$(pidof kubelet)/cwd
lrwxrwxrwx 1 root root 0 May  9 00:29 /proc/16481/cwd -> /
```

**Comment by mikedanese** [comment]
Are you running HEAD or a release version? There is a bit of logging around selecting a .dockercfg at the v4 logging level [here](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/credentialprovider/config.go#L66-L90). To verify it is an issue with kubelet not picking up the .dockercfg, can you start restart the kubelet with `--v=4` and grep the logs for `dockercfg` and share the results? Those log statements should happen within the first couple minutes of startup.

**Comment by bkeroackdsc** [comment]
@mikedanese I am running release version 0.15.0

I modified the kube-kublet unit as specified, restarted and at first did not get any dockercfg messages in the logs. I then manually removed and deleted all containers and images with the docker client locally on the minion and restarted kublet--I saw a msg saying something to the effect of "new credentials registered from .dockercfg".

Now image pulls succeed as expected. I'm not sure what happened (perhaps just restarting kubelet was sufficient?) but it appears to work now.
